src/main/java/jp/ikedam/jenkins/plugins/extensible_choice_parameter/DatabaseChoiceListProvider.java

```diff
@@ -0,0 +1,419 @@
+ * THE SOFTWARE.
+ */
+package jp.ikedam.jenkins.plugins.extensible_choice_parameter;
+
+import hudson.Extension;
+import hudson.model.Descriptor;
+import hudson.util.FormValidation;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.Serializable;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Scanner;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+/**
+ * Create a choice list by querying a specific column of a table in a database.
+ * 
+ * @author Thibault Duchateau
+ */
+public class DatabaseChoiceListProvider extends ChoiceListProvider implements Serializable
+{
+	private static final long serialVersionUID = 8488443485003760275L;
+        
+    private static final Logger LOGGER = Logger.getLogger(DatabaseChoiceListProvider.class.getName());
+    
+    private String jdbcUrl;
+    private String dbName;
+    private String dbTable;
+    private String dbColumn;
+    private String dbUsername;
+    private String dbPassword;
+    private String jdbcDriver;
+    private String fallbackFilePath;
+    
+	public String getDbUsername() {
+		return dbUsername;
+	}
+
+	public void setDbUsername(String dbUsername) {
+		this.dbUsername = dbUsername;
+	}
+
+	public String getDbPassword() {
+		return dbPassword;
+	}
+
+	public void setDbPassword(String dbPassword) {
+		this.dbPassword = dbPassword;
+	}
+
+	public String getJdbcUrl() {
+		return jdbcUrl;
+	}
+
+	public void setJdbcUrl(String jdbcUrl) {
+		this.jdbcUrl = jdbcUrl;
+	}
+	
+	public String getDbColumn() {
+		return dbColumn;
+	}
+
+	public void setDbColumn(String colomndb) {
+		this.dbColumn = colomndb;
+	}
+
+	public String getDbName() {
+		return dbName;
+	}
+
+	public void setDbName(String namedb) {
+		this.dbName = namedb;
+	}
+
+	public String getDbTable() {
+		return dbTable;
+	}
+
+	public void setDbTable(String tabledb) {
+		this.dbTable = tabledb;
+	}
+	
+	public String getJdbcDriver() {
+		return jdbcDriver;
+	}
+
+	public void setJdbcDriver(String driverdb) {
+		this.jdbcDriver = driverdb;
+	}
+
+	public String getFallbackFilePath() {
+		return fallbackFilePath;
+	}
+
+	public void setFallbackFile(String fallbackFilePath) {
+		this.fallbackFilePath = fallbackFilePath;
+	}
+
+	@Override
+	public List<String> getChoiceList() {
+		List<String> allChoices = performDbQuery();
+		LOGGER.log(Level.CONFIG, "Returned " + allChoices.size() + " entries from the database");
+		return allChoices;
+	}
+	
+	@DataBoundConstructor
+    public DatabaseChoiceListProvider(String userdb, String passworddb, String urldb, String colomndb, String namedb, String tabledb, String driverdb, String fallbackfile)
```

Comment:
This shouldn't work as Jenkins binds parameters in jelly and parameters in the constructer with their definition name.
(they should be `String dbUsername, String dbPassword, ...)